Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.cpp

```diff
@@ -112,6 +112,10 @@ PsatdAlgorithm::PsatdAlgorithm(
     {
         amrex::Abort("PSATD: warpx.do_divb_cleaning = 1 implemented only with psatd.J_linear_in_time = 1");
     }
+    if (time_averaging && !update_with_rho)
+    {
+        amrex::Abort("PSATD: warpx.time_averaging = 1 implemented only with psatd.update_with_rho = 1");
+    }
```

Comment:
As for how to do this, I'm thinking that I'd be more in favor of resetting `psatd.update_with_rho = 1` automatically and issuing a warning to the user, instead of aborting.

Something like this:
```
        if (fft_do_time_averaging == true && update_with_rho == false)
        {
            update_with_rho = true;
            RecordWarning(
                "PSATD",
                "Setting psatd.update_with_rho = 1 because psatd.do_time_averaging = 1",
                WarnPriority::low);
        }
```
which would go after these lines in WarpX.cpp (where we query the user's input for `psatd.update_with_rho`)
https://github.com/ECP-WarpX/WarpX/blob/f7ab88551c31537b625e12c16086c100544c5ae3/Source/WarpX.cpp#L1161-L1162

What do you and @RemiLehe think?

If you decide to keep the abort instead, we would need anyways to change `warpx.time_averaging = 1` with `psatd.do_time_averaging = 1` in the abort message.

Reply:
We could discuss this during the developer meeting. I think I have a slight preference for the current setting in the PR.